core/src/main/java/jenkins/monitor/EndOfLifeAdminMonitor.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2021 Tim Jacomb.
+ * Copyright 2023 Mark Waite.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.monitor;
+
+import static java.util.logging.Level.FINE;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.model.AdministrativeMonitor;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.time.LocalDate;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import jenkins.model.Jenkins;
+import jenkins.util.SystemProperties;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.DoNotUse;
+import org.kohsuke.stapler.HttpRedirect;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.HttpResponses;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+/* Package protected so that it won't be visible outside this package */
+class EndOfLifeAdminMonitor extends AdministrativeMonitor {
```

Comment:
I wonder if this should be just one admin monitor for Operating System End of Life.

you could still have it as an extension point so the logic could be separated but you wouldn't have all these pointless admin monitors in the enabled list as a controller is only ever going to run on one OS.

<img width="529" alt="image" src="https://user-images.githubusercontent.com/21194782/235467421-d64f6053-1b46-4a33-ae34-975e5b783812.png">

Reply:
I like that suggestion very much.  I'll need to investigate further